librz/analysis/rtti.c

```diff
@@ -14,30 +14,30 @@ RZ_API char *rz_analysis_rtti_demangle_class_name(RzAnalysis *analysis, const ch
 	return rz_analysis_rtti_itanium_demangle_class_name(&context, name);
 }
 
-RZ_API void rz_analysis_rtti_print_at_vtable(RzAnalysis *analysis, ut64 addr, int mode) {
-	bool use_json = mode == 'j';
+RZ_API void rz_analysis_rtti_print_at_vtable(RzAnalysis *analysis, ut64 addr, RzCmdStateOutput *state) {
+	bool use_json = state->mode == RZ_OUTPUT_MODE_JSON;
 	if (use_json) {
 		rz_cons_print("[");
 	}
 
 	RVTableContext context;
 	rz_analysis_vtable_begin(analysis, &context);
 	if (context.abi == RZ_ANALYSIS_CPP_ABI_MSVC) {
-		rz_analysis_rtti_msvc_print_at_vtable(&context, addr, mode, false);
+		rz_analysis_rtti_msvc_print_at_vtable(&context, addr, state, false);
 	} else {
-		rz_analysis_rtti_itanium_print_at_vtable(&context, addr, mode);
+		rz_analysis_rtti_itanium_print_at_vtable(&context, addr, state);
 	}
 
 	if (use_json) {
 		rz_cons_print("]\n");
 	}
 }
 
-RZ_API void rz_analysis_rtti_print_all(RzAnalysis *analysis, int mode) {
+RZ_API void rz_analysis_rtti_print_all(RzAnalysis *analysis, RzCmdStateOutput *state) {
 	RVTableContext context;
 	rz_analysis_vtable_begin(analysis, &context);
 
-	bool use_json = mode == 'j';
+	bool use_json = state->mode == 'j';
 	if (use_json) {
 		rz_cons_print("[");
 	}
```

Comment:
```suggestion
	rz_cmd_state_output_array_start(state);
```

Reply:
Similarly, use _end instead of printing `]` at the end of the function.

Reply:
For this PR, I don't want to change the other codes that are not related to the `newshell`...

Reply:
Because it's just a converting.

Reply:
It is related to the newshell. It helps us to avoid passing `PJ *pj` everywhere. And it's a small change, see this self-contained example: https://github.com/rizinorg/rizin/blob/dev/librz/core/chash.c

Reply:
I have changed the command type to RZ_CMD_DESC_TYPE_ARGV_MODES, and it works well. So there is no RzCmdStateOutput to use. Also, if use RzCmdStateOutput directly like the code I wrote before, it will break the test because the current print function is printing out by order, but the RzCmdStateOutput is to print out at the end, thus I need to change the logic of the current print function.